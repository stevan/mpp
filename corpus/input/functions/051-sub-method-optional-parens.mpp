# Subroutine without parentheses
sub foo {
    return 42;
}

# Subroutine with explicit empty parentheses (should be the same)
sub bar () {
    return 42;
}

# Edge case: subroutine with body but no params
sub no_params {
    my $x = 1;
    my $y = 2;
    return $x + $y;
}
