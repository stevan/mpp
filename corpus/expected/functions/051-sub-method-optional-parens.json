[
  {
    "type": "Sub",
    "name": "foo",
    "parameters": [],
    "body": [
      {
        "type": "Return",
        "value": {
          "type": "Number",
          "value": "42"
        }
      },
      {
        "type": "Sub",
        "name": "bar",
        "parameters": [],
        "body": [
          {
            "type": "Return",
            "value": {
              "type": "Number",
              "value": "42"
            }
          },
          {
            "type": "Sub",
            "name": "no_params",
            "parameters": [],
            "body": [
              {
                "type": "Declaration",
                "declarator": "my",
                "variable": {
                  "type": "Variable",
                  "name": "$x"
                },
                "initializer": {
                  "type": "Number",
                  "value": "1"
                }
              },
              {
                "type": "Declaration",
                "declarator": "my",
                "variable": {
                  "type": "Variable",
                  "name": "$y"
                },
                "initializer": {
                  "type": "Number",
                  "value": "2"
                }
              },
              {
                "type": "Return",
                "value": {
                  "type": "Variable",
                  "name": "$x"
                }
              }
            ]
          }
        ]
      }
    ]
  }
]